src/CloudStore.jl

```diff
@@ -69,6 +69,38 @@ head(x::Azure.Container, key::String; kw...) = Blobs.head(x, key; kw...)
 put(x::Azure.Container, key::String, in::RequestBodyType; kw...) = Blobs.put(x, key, in; kw...)
 delete(x::Azure.Container, key::String; kw...) = Blobs.delete(x, key; kw...)
 
+# https://docs.aws.amazon.com/AmazonS3/latest/userguide/bucketnamingrules.html
+function validate_bucket_name(bucket_name, accelerate)
+    !(3 <= length(bucket_name) <= 63) && error("Invalid bucket name $(repr(bucket_name)): Bucket names must be between 3 (min) and 63 (max) characters long.")
+    isnothing(match(r"[a-z0-9][\.\-a-z0-9]+[a-z0-9]", bucket_name)) && error("Invalid bucket name $(repr(bucket_name)): Bucket names can consist only of lowercase letters, numbers, dots (.), and hyphens (-). Bucket names must begin and end with a letter or number.")
+    occursin("..", bucket_name) && error("Invalid bucket name $(repr(bucket_name)): Bucket names must not contain two adjacent periods.")
+    split(bucket_name, '.') == 4 && all(!isnothing, tryparse(Int, s) for s in split(bucket_name, '.')) && error("Invalid bucket name $(repr(bucket_name)): Bucket names must not be formatted as an IP address (for example, 192.168.5.4)")
+    startswith(bucket_name, "xn--") && error("Invalid bucket name $(repr(bucket_name)): Bucket names must not start with the prefix `xn--`.")
+    endswith(bucket_name, "-s3alias") && error("Invalid bucket name $(repr(bucket_name)): Bucket names must not end with the suffix `-s3alias`.")
+    accelerate && occursin(".", bucket_name) && error("Invalid bucket name $(repr(bucket_name)): Buckets used with Amazon S3 Transfer Acceleration can't have dots (.) in their names.")
+    return bucket_name
+end
```

Comment:
We should also unit test these validation functions with a bunch of examples! This stuff is really tricky to get right, so having tons of examples in the tests will be really helpful